huntsman/observatory.py

```diff
@@ -476,6 +481,22 @@ def prepare_cameras(self, sleep=60, max_attempts=5, require_all_cameras=False):
                                   f'waiting another {sleep} seconds before checking again.')
                 time.sleep(sleep)
 
+        # Handle failed cameras
+        if len(cameras_failed) != 0:
+            msg = "The following cameras have failed to become ready: "
+            msg += ", ".join(cameras_failed)
+
+            # Raise PanError if we need all cameras
+            if require_all_cameras:
+                raise error.PanError(msg)
+
+            # Drop the camera if we don't need all cameras
+            else:
+                msg += ". Dropping them from the Observatory."
```

Comment:
The continued `msg` is a bit weird to me. I would just have the message in the `PanError` above and not have this part of the message or have it as a separate log call.